23.1.0: pytest freezess on collection units

I'm trying to package your module as an rpm package. So I'm using the typical PEP517 based build, install and test cycle used on building packages from non-root account.
- `python3 -sBm build -w --no-isolation`
- because I'm calling `build` with `--no-isolation` I'm using during all processes only locally installed modules
- install .whl file in </install/prefix>
- run pytest with PYTHONPATH pointing to sitearch and sitelib inside </install/prefix>

Looks like pytest is feezing on collecting units:
<details>

```console
+ PYTHONPATH=/home/tkloczko/rpmbuild/BUILDROOT/python-zmq-23.1.0-3.fc35.x86_64/usr/lib64/python3.8/site-packages:/home/tkloczko/rpmbuild/BUILDROOT/python-zmq-23.1.0-3.fc35.x86_64/usr/lib/python3.8/site-packages
+ /usr/bin/pytest -ra --import-mode=importlib
==================================================================== test session starts =====================================================================
platform linux -- Python 3.8.13, pytest-7.1.2, pluggy-1.0.0
rootdir: /home/tkloczko/rpmbuild/BUILD/pyzmq-23.1.0, configfile: pytest.ini
collecting ... 
```
</details>

Test env it is just spawned LXC zone in which ar installed only packages listed in rpm BuildRequires.
```console
+ /usr/bin/pytest -ra --import-mode=importlib --pdb
==================================================================== test session starts =====================================================================
platform linux -- Python 3.8.13, pytest-7.1.2, pluggy-1.0.0
rootdir: /home/tkloczko/rpmbuild/BUILD/pyzmq-23.1.0, configfile: pytest.ini
collecting ... ^C
[tkloczko@devel-g2v SPECS]$
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> traceback >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
mypy_tests/test_context.py:5: in <module>
    s.send(b"buf")
../../BUILDROOT/python-zmq-23.1.0-3.fc35.x86_64/usr/lib64/python3.8/site-packages/zmq/sugar/socket.py:618: in send
    return super().send(data, flags=flags, copy=copy, track=track)
zmq/backend/cython/socket.pyx:740: in zmq.backend.cython.socket.Socket.send
    cpdef send(self, object data, int flags=0, copy=True, track=False):
zmq/backend/cython/socket.pyx:787: in zmq.backend.cython.socket.Socket.send
    return _send_copy(self.handle, data, flags)
zmq/backend/cython/socket.pyx:244: in zmq.backend.cython.socket._send_copy
    _check_rc(rc)
zmq/backend/cython/checkrc.pxd:13: in zmq.backend.cython.checkrc._check_rc
    PyErr_CheckSignals()
E   KeyboardInterrupt
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> entering PDB >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB post_mortem (IO-capturing turned off) >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
> /home/tkloczko/rpmbuild/BUILD/pyzmq-23.1.0/zmq/backend/cython/checkrc.pxd(13)zmq.backend.cython.checkrc._check_rc()
-> PyErr_CheckSignals()
(Pdb)
```

--- a/pytest.ini
+++ b/pytest.ini
@@ -3,3 +3,5 @@ markers =
     large: these tests use a lot of memory
     new_console: these tests create a new console
     wheel: these tests are for installs from a wheel, not dev-installs
+testpaths =
+    zmq/tests

